shell/platform/embedder/embedder.h

```diff
@@ -611,6 +611,36 @@ typedef struct {
   int64_t buttons;
 } FlutterPointerEvent;
 
+typedef enum {
+  kFlutterKeyEventKindUp = 1,
+  kFlutterKeyEventKindDown,
+  kFlutterKeyEventKindRepeat,
+} FlutterKeyEventKind;
+
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterKeyEvent).
+  size_t struct_size;
+  // Timestamp in microseconds.
+  double timestamp;
+  // The event kind.
+  FlutterKeyEventKind kind;
+  // The physical key of the event, distinguished by the HID code.
+  uint64_t physical;
+  // The logical key of the event, usually the effect of the key without
+  // modifier, but might include modifier if the information is not available.
+  // Can be 0 for empty.
+  uint64_t logical;
+  // Null-terminated character input from the event. Can be null. Not available
+  // to up events.
+  const char* character;
+  // Whether the event is synthesized by Flutter.
```

Comment:
This comment is very confusing, since the embedder is providing this data, and it's not Flutter. It's whether it is synthesized by the embedder. And as with the above, this comment needs a lot more context: what is a synthesized event, and when should an embedder send one?